rules.js

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright 2024 [TODO: COPYRIGHT HOLDER NAME]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// rules.js
+
+import { urlFilterParse, urlMatcher } from './urlFilterParser.js';
+import { rulesetFilePaths } from './manifest.js';
```

Comment:
I think it might be a larger change, so no need to do anything here, but I'm uncertain if importing state from other files like this is a good idea. It might be nice to find an alternative at some point.

Reply:
How about localStorage or events?

Reply:
Is the plan to eventually extract these from an upload of the entire extension? If so, maybe when we do that we can remove this input entirely and just call the function with the rulesets we extract?